### ISSUE TITLE
Fix side tree panel accessibility in manager interface, multiple issues (accessibility defect)

### ISSUE THREAD

**Issue by paulbohman** [modxcms/revolution#12153]
For compliance with Web Content Accessibility Guidelines (WCAG)

Keyboard accessibility of tablist/tab panels:

The tab panel is not keyboard accessible at all. The main tabs need to have tabindex="0" on the active tab and tabindex="-1" on the inactive tabs, and the JavaScript needs to toggle that value as users select different tabs. You also need to toggle aria-expanded="true" and aria-controls="the-id-of-the-tabpanel". See https://dequeuniversity.com/assets/js/aria/saveform/form2.html for an example.

Icon alternative text:

All of the icons (New document, New symlink, etc. and Trash) need alt text so screen readers know what to say. See "alt text for images and icons" in the uberbar above for possible techniques.

Button role for icons:

The icons (New document, New symlink, etc. and Trash) should be given role=&quot;button&quot; or they should be enclosed in actual &lt;button&gt; tags because they are perform application-type features, and a button is the appropriate semantic role to convey to screen reader users.

Keyboard-accessible tooltips for icon/buttons:

The tooltip should popup when the icons/buttons receive keyboard focus, and not just with mouseover events.

Keyboard accessibility of the button to collapse side bar:

The handle to collapse the entire side bar to the left needs to be keyboard accessible, and it needs to be a &lt;button&gt; or have role=&quot;button&quot; and it must have a label (&quot;collapse sidebar&quot; for example) to convey to users its purpose.

Keyboard accessibility of tree view:

Users need to be able to access all items and functionality (refresh, add new __ here, etc.) in the tree view with the keyboard. It would be best to impelement an ARIA tree menu that allows navigation with the arrow key. You would tab into the tree object and then use the arrow keys to navigate within it, including expanding and collapsing nodes in the tree. Using the tab key would take you out of the tree, not navigate within it. Tree views are kind of complicated, but they can be made accessible. You would need to add ARIA attributes and toggle between various states such as aria-expanded="true". You would also need to make sure that the functionality of the expand/collapse icon/triangle is clear, and that it is separate from the functionality of the page title itself, which opens the page. 
See http://www.w3.org/TR/wai-aria/roles#tree
http://accessibleculture.org/articles/2013/02/not-so-simple-aria-tree-views-and-screen-readers/

Keyboard accessibility of right-click menu in the tree view:

The functionality of the right-click menu needs to be available to keyboard users. All of the features are available other places, so this requirement is sort of satisfied already, but it would be best to make the right-click menu directly available to keyboard users, rather than make them go through the less efficient routes to the same features. You could implement a keyboard shortcut such as alt+enter on links in the menu to activate the right-click menu. To be effective, though, you would need to tell users that this keyboard shortcut exists, so you could add a tooltip to each link saying "Use alt + enter for contextual menu". You would need to ensure that screen readers also read this text. Particularly important is the ability to add new items (resources, elements, files) as children of the item that currently has the focus, since this is a very common task.

Keyboard focus management:

When you select a page to edit it, the focus initially goes to the pagetitle field, which seems logical. I like that. The problem is that the menu tree loads and refreshes, and as it does so, it eventually steals the keyboard focus away from that field and puts the focus on the current page in the tree view. This breaks the keyboard flow, and if you can't use a mouse, you have to tab, tab, tab, tab, tab, a lot of times to get back to where you were. Ensure that the tree view doesn't steal the focus.

See accessibility to-do list at https://dequeuniversity.com/modx-accessibility.html#tree-panel

**Comment by rthrash** [comment]
This is being addressed at the [MODX Accessible CMS Project](https://github.com/modxcms/a11y/issues/), and with Paul's agreement has been moved there. For more details, see https://github.com/modxcms/revolution/issues/12156#issuecomment-72382480. Thank you!
